dateutil/parser.py

```diff
@@ -1328,9 +1327,6 @@ def _parse_hms(i, l, info, res):
     return i
 
 
-# TODO: require len(token) >= 3 like we do for the between-parens version?
```

Comment:
OK.  There are two things I'd like to accomplish here (not necessarily in this PR):

1) Get as much of the could-this-be-a-tzname logic into one place.  So if `Z` is allowed but aside from that special case there are length-based criteria, that would be easy to put into `_could_be_tzname`.  You've put much more though than I have into timezone stuff, so I'm going to defer to you on what this logic should actually be.

2) Unify/merge the logic used here and in the between-parens tzname criteria.  Or if it can't be unified, at least clearly document the relevant differences.